tower-http/src/timeout/body.rs

```diff
@@ -0,0 +1,379 @@
+//! Middleware that applies a timeout to bodies.
+//!
+//! Bodies must produce data at most within the specified timeout.
+//! If they are inactive, an error will be generated.
+//!
+//! # Differences from `tower_http::timeout::service::Timeout`
+//!
+//! `tower_http::timeout::service::Timeout` applies a timeout on the full request.
```

Comment:
I would maybe say something like it applies the timeout to the request future and not the body. Bodies are handled asynchronously outside of the tower stack's future and thus needs an additional timeout.